src/main/java/org/snomed/snowstorm/core/data/services/CodeSystemService.java

```diff
@@ -577,14 +577,28 @@ public void updateCodeSystemBranchMetadata(CodeSystem codeSystem, CodeSystemNewA
 			throw new IllegalStateException("Branch metadata can not be updated when branch is locked.");
 		}
 		Metadata branchMetadata = branch.getMetadata();
-		if (updateRequest.getPreviousPackage() != null) {
-			branchMetadata.putString(PREVIOUS_PACKAGE, updateRequest.getPreviousPackage());
-		}
 		if (updateRequest.getPreviousRelease() != null) {
 			branchMetadata.putString(PREVIOUS_RELEASE, updateRequest.getPreviousRelease());
+
+			// Get previous package
+			final CodeSystemVersion codeSystemVersion = findVersion(codeSystem.getShortName(), Integer.parseInt(updateRequest.getPreviousRelease()));
+			if (codeSystemVersion == null) {
+				throw new IllegalStateException("Previous release " + updateRequest.getPreviousRelease() + " not found.");
+			}
+			branchMetadata.putString(PREVIOUS_PACKAGE, codeSystemVersion.getReleasePackage());
 		}
-		if (updateRequest.getPreviousDependency() != null) {
-			branchMetadata.putString(PREVIOUS_DEPENDENCY_PACKAGE, updateRequest.getPreviousDependency());
+		if (updateRequest.getPreviousDependencyRelease() != null && codeSystem.getDependantVersionEffectiveTime() != null) {
```

Comment:
@astro-snail  It looks good. After checking it further, we can derive previousDependencyRelease from the CodeSystemVersion using the value of dependantVersionEffectiveTime. Therefore we can remove the prevousDependencyRelease parameter. I will update the ticket for this. Thanks.